compiler/runtime/OMRRuntimeAssumptions.hpp

```diff
@@ -185,6 +197,77 @@ class PatchNOPedGuardSite : public OMR::LocationRedirectRuntimeAssumption
    uint8_t *_destination;
    }; // TR::PatchNOPedGuardSite
 
+class PatchSites
+   {
+   private:
+   size_t    _refCount;
+   size_t    _size;
+   size_t    _maxSize;
+
+   /**
+    * List of location and destination pairs, eg. {loc0, dest0, loc1, dest1, ...}
+    * This improves locality and reduces memory overhead.
+    */
+   uint8_t **_patchPoints;
+
+   // Cache lower and upper bounds
+   uint8_t *_firstLocation;
+   uint8_t *_lastLocation;
+
+   public:
+   PatchSites(TR_PersistentMemory *pm, size_t maxSize);
+
+   size_t   getSize() { return _size; }
+   uint8_t *getLocation(size_t index);
+   uint8_t *getDestination(size_t index);
+   uint8_t *getFirstLocation() { return _firstLocation; }
+   uint8_t *getLastLocation() { return _lastLocation; }
+
+   void add(uint8_t *location, uint8_t *destination);
+
+   bool equals(PatchSites *other);
+   bool containsLocation(uint8_t *location);   
```

Comment:
Is this called publically? If so do we need to add fast paths to this like equals and have an internal implementation that is used by both equals and this one to save scanning? Otherwise this can be the private/protected helper...

Reply:
At the moment, its not called publically, but it could provide useful functionality. An internal helper has been added.